packages/react-component-library/src/components/RangeSlider/ThresholdTrack.tsx

```diff
@@ -0,0 +1,81 @@
+import React from 'react'
+import classNames from 'classnames'
+import { TrackItem, GetTrackProps } from 'react-compound-slider'
+
+interface ThresholdTrackProps extends TrackItem {
+  getTrackProps: GetTrackProps
+  thresholds?: number[]
+}
+
+interface ChunkProps extends ComponentWithClass {
+  left: number
+  width: number
+  maxWidth: number
+  testId: string
+}
+
+export const ThresholdTrack: React.FC<ThresholdTrackProps> = ({
+  target,
+  getTrackProps,
+  thresholds,
+}) => {
+  const singleThreshold = thresholds?.length === 1
+  const doubleThreshold = thresholds?.length === 2
+
+  const Chunk: React.FC<ChunkProps> = ({
```

Comment:
It feels like this deviates away from current patterns which comes at the cost of making the `ThresholdTrack` `function` less readable.

Reply:
Hmm yeah you think move it out into a module of its own?

Reply:
I remember now there was actually a reason for me including it in here - if I remove it out into a sub-component outside of the the block is a re-render bug as you move the slider left and right. I'll make a gif and show you.

Reply:
![2020-06-19 15 39 24](https://user-images.githubusercontent.com/48086589/85144509-274afc00-b243-11ea-9ff3-ff23cae158df.gif)